Automattic/jetpack
# tests: Replace deprecated PHPUnit `assertObject(Not)HasAttribute`

## Proposed changes:

PHPUnit 10 at first removed this but then in 10.1 re-added it as `assertObject(Not)HasProperty`. Now that a replacement exists, yoast/phpunit-polyfills provides a polyfill, so let's make use of it.

Note WordPress's `WP_UnitTestCase` and WorDBless's `WorDBless\BaseTestCase` both already include all Yoast polyfill traits, so explicit use of the trait is only needed for things using PHPUnit's base class.

### Other information:


## Jetpack product discussion


None

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Is CI happy?
* Check the PHP 8.2 WP latest run. Are there any of the deprecation notices left?
